### ISSUE TITLE
Variable values not persisted on application force-quit/crash

### ISSUE THREAD

**Issue by hugecannon** [riok/Kreya#249]
**Describe the bug**
Variable updates are missed if Kreya is force-quit on Mac

**To Reproduce**
1. Set up a request with a script which stores some variable after completion
2. Use said variable in request (e.g. query params) to show what state Kreya knows the variable to be
3. Run a request to ensure variable is set
4. Force quit Kreya
5. Re-open Kreya, variable has gone back to previous version (i.e. not latest)

**Expected behavior**
Variable state should be preserved whenever possible, i.e. latest updates to variables should be durable

**Screenshots**

**Environment (if possible, copy the information from the error dialog or the About menu):**
 - OS: MacOS (arm)
 - Kreya Version latest

**Additional context**
It seems possible variable state is only being written on application close, perhaps there should be some more regular routine sync/save.

**Comment by CommonGuy** [comment]
Yes, currently Kreya only saves the variable state on exiting. We will check whether we can change this to saving immediately.
